File: 331. Verify Preorder Serialization of a Binary Tree/Python/Solution.py
class Solution(object):
    def isValidSerialization(self, preorder):
        """
        :type preorder: str
        :rtype: bool
        """
        degree = -1
        for node in preorder.split(','):
            degree += 1
            if degree > 0:
                return False
            if node != '#':
                degree -= 2
        return degree == 0

File: 331. Verify Preorder Serialization of a Binary Tree/Java/Solution.java
public class Solution {
    public boolean isValidSerialization(String preorder) {
        int degree = -1;
        String[] nodes = preorder.split(",");
        for (String node : nodes) {
            degree++;
            if (degree > 0) {
                return false;
            }
            if (!node.equals("#")) {
                degree -= 2;
            }
        }
        return degree == 0;
    }
}